app/src/main/java/com/randomappsinc/studentpicker/models/ListInfo.java

```diff
@@ -90,6 +90,19 @@ public void removeNames(String name, int amount) {
         }
     }
 
+    public void setNameAmount(String name, int newAmount) {
+        if (nameAmounts.containsKey(name)) {
+            int currentAmount = nameAmounts.get(name);
+            if (currentAmount > newAmount) {
```

Comment:
At this level, we shouldn't invoke other APIs since that will lead to duplicated code. We can just update the HashMap directly. Make sure to keep numInstances updated properly though.